### PR TITLE
use float input as str for decimal type

### PR DESCRIPTION
**Pull request by jennifersp**
The scale of float input in INSERT statement gets rounded when converting to float64 value.
The scale of input for DECIMAL data type should not be rounded.

### CODE REVIEW

File: sql/parse/parse.go [hunk 1]
```diff
@@ -3381,6 +3381,16 @@ func convertVal(ctx *sql.Context, v *sqlparser.SQLVal) (sql.Expression, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		// use the value as string format to keep precision and scale as defined for DECIMAL data type to avoid rounded up float64 value
+		if ps := strings.Split(string(v.Val), "."); len(ps) == 2 {
+			if scale, err := strconv.ParseUint(ps[1], 10, 64); err == nil && scale > 0 {
```
Comment (fulghum) [inline sql/parse/parse.go:3387]: Does this work for decimal types with even more precision digits? I'm wondering if this will return an error for a really long decimal component. It would be good to add a test for the max number of decimal digits and make sure we still parse it correctly.

**Review by fulghum — COMMENTED** [review]
Change looks good; just a question about some boundary cases to make sure it scales to the max precision of a decimal. 

Also... could we unit test any of this behavior in `parse_test.go`?

File: sql/parse/parse.go (hunk 1 above)
Comment (jennifersp) [reply sql/parse/parse.go]: Anytime decimal value bigger than specified precision and scale, it creates warning and uses rounded value.

File: enginetest/queries/script_queries.go [hunk 2]
```diff
@@ -1610,6 +1610,47 @@ var ScriptTests = []ScriptTest{
...
+			{
+				Query:    "INSERT INTO test VALUES (119815923291839784651.11981592329183978465111981592329183978465144);",
+				Expected: []sql.Row{{sql.NewOkResult(1)}},
+			},
+			{
+				Query:    "SELECT COUNT(*) FROM test WHERE number = CONVERT('119815923291839784651.1198159232918398', DECIMAL)",
+				Expected: []sql.Row{{1}},
+			},
+			{
+				Query:          "create table invalid_decimal (number decimal(65,31));",
```
Comment (fulghum) [inline enginetest/queries/script_queries.go:1641]: I think we have these three error cases for invalid decimal fields tested in other places. Do a quick double check and confirm you don't need to retest these here.

File: enginetest/queries/script_queries.go [hunk 3]
```diff
@@ -1610,6 +1610,47 @@ var ScriptTests = []ScriptTest{
...
+			{
+				Query:    "INSERT INTO test VALUES (11981.5923291839784651);",
+				Expected: []sql.Row{{sql.NewOkResult(1)}},
+			},
+			{
+				Query:    "SELECT COUNT(*) FROM test WHERE number = CONVERT('11981.5923291839784651', DECIMAL)",
+				Expected: []sql.Row{{2}},
+			},
+			{
+				Query:    "INSERT INTO test VALUES (119815923291839784651.11981592329183978465111981592329183978465144);",
```
Comment (fulghum) [inline enginetest/queries/script_queries.go:1633]: Nice! Good test for a boundary condition. You should try moving all the allowed digits to the right of the decimal point so that we can really stress test our handling of the decimal portion, and you should add a check to make sure we get the correct expected value back without rounding.

File: sql/decimal.go [hunk 4]
```diff
@@ -84,15 +84,16 @@ var InternalDecimalType DecimalType = decimalType{
 
 // CreateDecimalType creates a DecimalType.
 func CreateDecimalType(precision uint8, scale uint8) (DecimalType, error) {
+	if scale > DecimalTypeMaxScale {
+		return nil, fmt.Errorf("Too big scale %v specified. Maximum is %v.", scale, DecimalTypeMaxScale)
```
Comment (fulghum) [inline sql/decimal.go:88]: Thank you for matching the MySQL error messages! 😄
